cakephpsphinx/themes/cakephp/static/js/app.js

```diff
@@ -0,0 +1,156 @@
+     * @property {string} search.highlightPostTag
+     * @property {string} search.encoder
+     * @property {App~Search~TemplatesConfig} search.templates
+     * @property {Object} standaloneSearch
+     * @property {number} standaloneSearch.limit
+     */
+    var _config = {};
+
+    var _defaultConfig = {
+        search: {
+            templates: {
+                instructions: function () {
+                    return App.Messages.substituteMessages(
+                        '<div class="instructions">' +
+                            '<p class="intro">{{search.instructions.intro}}</p>' +
+                            '<ul class="syntax">' +
+                                '<li>{{search.instructions.phrases}}</li>' +
+                                '<li>{{search.instructions.modifiers}}</li>' +
+                                '<li>{{search.instructions.boolean}}' +
+                            '</ul>' +
+                            '<p class="notes">{{search.instructions.notes}}</p>' +
+                        '</div>'
+                    );
+                },
+
+                pending: function () {
+                    return App.Messages.substituteMessages(
+                        '<div class="status pending">{{search.pending}}</div>'
+                    );
+                },
+
+                empty: function () {
+                    return App.Messages.substituteMessages(
+                      '<div class="status empty">{{search.empty}}</div>'
+                    );
+                },
+
+                /**
+                 * @param {string} type
+                 * @returns {string}
+                 */
+                error: function (type) {
+                    return App.Messages.substituteMessages(
+                      '<div class="status error">{{search.error.' + type + '}}</div>'
+                    );
+                },
+
+                /**
+                 * @param {App~Search~Result} result
+                 * @returns {string}
+                 */
+                result: function (result) {
+                    var $link = $('<a class="result"></a>');
+                    $link.attr('href', result.url);
+
+                    var $article = $('<article></article>');
+                    $link.append($article);
+
+                    var $header = $('<header></header>');
+                    $article.append($header);
+
+                    var hierarchy = result.hierarchy.slice();
+                    if (result.highlights.hierarchy.length) {
+                        hierarchy = result.highlights.hierarchy;
+                    }
+
+                    var section = hierarchy.shift();
+                    $header.append($('<h1 class="section"></h1>').html(section));
+
+                    if (hierarchy.length) {
+                        var title = '> ' + hierarchy.join(' > ');
+                        $header.append($('<p class="title"></p>').html(title));
```

Comment:
Do you need `.html()` here or would `.text()` work?

Reply:
No, `html()` is now actually needed. I've enabled Elasticsearch's tag highlighting, because a) I didn't wanted to deal with re-implementing highlighting in JS (the typeahead.js one was already borked), and I b) wanted incomplete words and typos to be highlighted too.

The data will be HTML encoded, except for the highlight tags generated by Elastic, so this should be safe.